pkg/antctl/raw/test/command.go

```diff
@@ -0,0 +1,522 @@
+	return &k8sConnectivityParams{
+		client:          client,
+		antreaNamespace: "kube-system",
+		params:          p,
+	}
+}
+
+func (k *k8sConnectivityParams) validatePodToPod(ctx context.Context) {
+	//conducts Pod to Pod connectivity tests within same node and different node
+	for _, clientPod := range k.clientPods.Items {
+		for echoName, echoIP := range k.echoPods {
+			var (
+				srcPod  = connectivityCheckNamespace + "/" + clientPod.Name
+				dstPod  = connectivityCheckNamespace + "/" + echoName
+				success = true
+			)
+			k.Header("Validating from pod %s to pod %s...", srcPod, dstPod)
+			_, err := k.client.ExecInPod(ctx, connectivityCheckNamespace, clientPod.Name, "", agnhostConnectCommand(echoIP+":80"))
+			if err != nil {
+				k.Log("curl connectivity check command failed: %s", err)
+				success = false
+			}
+
+			if success {
+				k.Log("client pod %s was able to communicate with echo pod %s (%s)", clientPod.Name, echoName, echoIP)
+			} else {
+				k.Log("client pod %s was not able to communicate with echo pod %s (%s)", clientPod.Name, echoName, echoIP)
+			}
+
+			k.Relax()
+		}
+	}
+}
+
+func (k *k8sConnectivityParams) validatePodInternetConnectivity(ctx context.Context) error {
+	for _, clientPod := range k.clientPods.Items {
+		var (
+			srcPod  = connectivityCheckNamespace + "/" + clientPod.Name
+			success = true
+		)
+
+		k.Header("Validating connectivity from pod %s to the world (google.com)...", srcPod)
+		_, err := k.client.ExecInPod(ctx, connectivityCheckNamespace, clientPod.Name, clientDeploymentName, agnhostConnectCommand("google.com:80"))
+		if err != nil {
+			k.Log("Connectivity test from pod %s to google.com failed: %s", srcPod, err)
+			success = false
+		}
+
+		if success {
+			k.Log("Pod %s was able to connect to google.com", srcPod)
+		}
+
+		k.Relax()
+	}
+	return nil
+}
+
+func (k *k8sConnectivityParams) Relax() {
+	time.Sleep(2 * time.Second)
+}
+
+type Parameters struct {
+	antreaNamespace string
+	SingleNode      bool
+	ForceDeploy     bool
+	Writer          io.Writer
+}
+
+const podReadyTimeout = 5 * time.Minute
+
+func (k *k8sConnectivityParams) deleteDeployments(ctx context.Context, client k8sClientOperations) error {
+	k.Log("[%s] Deleting connectivity check deployments...", client.ClusterName())
+	client.DeleteDeployment(ctx, connectivityCheckNamespace, echoSameNodeDeploymentName, metav1.DeleteOptions{})
+	client.DeleteDeployment(ctx, connectivityCheckNamespace, echoOtherNodeDeploymentName, metav1.DeleteOptions{})
+	client.DeleteDeployment(ctx, connectivityCheckNamespace, clientDeploymentName, metav1.DeleteOptions{})
+	client.DeleteService(ctx, connectivityCheckNamespace, echoSameNodeDeploymentName, metav1.DeleteOptions{})
+	client.DeleteService(ctx, connectivityCheckNamespace, echoOtherNodeDeploymentName, metav1.DeleteOptions{})
+	client.DeleteNamespace(ctx, connectivityCheckNamespace, metav1.DeleteOptions{})
```

Comment:
Since everything is created in the same Namespace, we can just delete the Namespace and it should delete everything inside the Namespace. That would be less verbose. I don't think there should be any issue with that.

Reply:
sure :)